---
title: "Analyzing Mutation Types with mutationTypR"
author: "Sabrina A. Tidjani"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Analyzing Mutation Types with mutationTypR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r style, echo = FALSE, results = 'asis'}
library(BiocStyle)
```

```{r, echo = FALSE}
library(knitr)
```
```{r, echo = FALSE}
library(mutationTypR)
```
# Introduction

The mutationTypR package provides tools for extracting and analyzing mutation types from VCF data. It focuses on identifying single nucleotide variants (SNVs) and categorizing them based on their genomic context and mutation characteristics.

# Usage

## Load necessary libraries

```{r}
library(VariantAnnotation)
library(GenomeInfoDb)
library(BSgenome.Hsapiens.UCSC.hg19)
```

## Example VCF file and reference genome (replace with actual file path or VCF object and actual reference genome object)

```{r}
vcffile <- system.file("extdata", "chr22.vcf.gz", package = "VariantAnnotation")
vcf_data <- VariantAnnotation::readVcf(vcffile, genome = "hg19")
vcf_data <- GenomeInfoDb::renameSeqlevels(vcf_data, 'chr22')
Hsapiens <- BSgenome.Hsapiens.UCSC.hg19
```

## Extract mutation types

```{r}
mutation_types <- extractMutationTypes(vcf_data, Hsapiens, 3)
```
# Summarize and plot mutation types

```{r}
summary_df <- summarizeAndPlotMutations(vcf_data, Hsapiens, context_length = 3, plot = TRUE, top_n = 20)
print(summary_df)
```

